apps/web/components/GardenCard.tsx

```diff
@@ -1,62 +1,65 @@
 "use client";
 
+import { BuildingStorefrontIcon } from "@heroicons/react/24/outline";
 import Image from "next/image";
-import Link from "next/link";
+import { Address } from "viem";
 import { getTokenGardensQuery } from "#/subgraph/.graphclient";
-import { Button } from ".";
+import { Statistic, TokenLabel } from ".";
 import { gardenLand } from "@/assets";
-import { ChainIcon, getChain } from "@/configs/chainServer";
+import { Card } from "@/components/Card";
+import { getChain } from "@/configs/chainServer";
 
 type TokenGarden = getTokenGardensQuery["tokenGardens"][number];
 
 export function GardenCard({ garden }: { garden: TokenGarden }) {
-  const { id, name, symbol, chainId } = garden;
+  const { id, name, chainId } = garden;
   const link = `/gardens/${chainId}/${id}`;
   const communities = garden.communities?.filter((comm) => comm.isValid);
   const commLength = communities?.length ?? 0;
-  const totalMembers =
-    communities
-      ?.map((comm) => comm.members?.length ?? 0)
-      .reduce((a, b) => a + b, 0) ?? 0; //@todo temporary, that can be take from the subgraph
+
+  // TODO: improve the types ?
+  function countUniqueMemberAddresses(communitiesArray: any[] | undefined) {
+    const uniqueAddresses = new Set();
+
+    communitiesArray?.forEach(community => {
+      if (community.members && Array.isArray(community.members)) {
+        community.members.forEach((member: { memberAddress: Address; }) => {
+          if (member.memberAddress) {
+            uniqueAddresses.add(member.memberAddress);
+          }
+        });
+      }
+    });
+    return uniqueAddresses.size;
+  }
+
+  const totalMembers = countUniqueMemberAddresses(communities);
+
   return (
-    <div className="border2 bg-surface relative flex max-w-[320px] flex-col overflow-hidden rounded-lg border-black shadow">
-      <div className="flex flex-col gap-2 p-2">
-        <div className="card relative" />
-        <div className="flex flex-col items-center gap-1">
-          <h4 className="text-2xl font-bold">{name}</h4>
-          <p className="text-center">{symbol}</p>
-          <div className="align-center flex justify-center">
-            <p className="text-center text-xs">Network:</p>
-            <div className="mx-1 content-center justify-center">
-              <ChainIcon chain={chainId} height={16} />
+    <Card href={link} >
+      <div className="flex flex-col gap-7">
+        <div className="flex items-center justify-center gap-2">
+          {/* TODO: find appropiate token image */}
+          <TokenLabel chainId={chainId} noSymbol />
```

Comment:
i was about to say find better token images but i saw the TODO comment, also it would be good to have it bigger what do you think?